stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -118,17 +121,17 @@ size(A::HermOrSym) = size(A.data)
     @inbounds if (A.uplo == 'U') == (i < j)
         return A.data[i, j]
     else
-        return A.data[j, i]
+        return transpose(A.data[j, i])
     end
 end
 @inline function getindex(A::Hermitian, i::Integer, j::Integer)
     @boundscheck checkbounds(A, i, j)
     @inbounds if (A.uplo == 'U') == (i < j)
         return A.data[i, j]
     elseif i == j
-        return eltype(A)(real(A.data[i, j]))
+        return convert(typeof(A.data[i, j]), real(A.data[i, j]))
```

Comment:
Ah, this is probably wrong if `A.data[i, j]` is a matrix: should probably rather be `Hermitian(...)` then..